airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -74,7 +90,27 @@ def conforms_to_schema(record: Mapping[str, Any], schema: Mapping[str, str]) ->
     - For every column in the record, that column's type is equal to or narrower than the same column's
       type in the schema.
     """
-    ...
+    schema_columns = set(schema.get("properties", {}).keys())
+    record_columns = set(record.keys())
+
+    if not record_columns.issubset(schema_columns):
+        return False
+
+    for column, definition in schema.get("properties", {}).items():
+        expected_types = definition.get("type", [])
+        value = record.get(column)
+
+        if value is not None:
+            if "object" in expected_types if isinstance(expected_types, list) else expected_types == "object":
```

Comment:
nit: these double-inlined-ifs are difficult to parse. what do you think about
`if (isinstance(expected_type, list) and "object" in expected_types) or expected_types == "object"`?